httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/HttpClientBuilder.java

```diff
@@ -838,6 +839,11 @@ protected Function<HttpContext, HttpClientContext> contextAdaptor() {
     }
 
     public CloseableHttpClient build() {
+
```

Comment:
@arturobernalg I am a bit unsure HttpClient should be meddling with system variables behind the scene. This can really have unexpected side-effects. Could we document `EnvironmentProxyConfigurer` in javadocs and advise the users to explicitly call `EnvironmentProxyConfigurer#apply` in order to carry over proxy config from the runtime environment?

Reply:
@ok2c  Oleg, one last thought before I push: would you prefer we gate the transfer behind a dedicated env-flag, e.g. `hc5_apply_proxy_env=true` so users who really want the mapping can simply set it and skip the
explicit `EnvironmentProxyConfigurer.apply()` call?
Default remains “do nothing unless asked”

Reply:
@arturobernalg I do not have a use case, so I can make this call. _Personally_ I would prefer just an explicit `EnvironmentProxyConfigurer#apply` call, but I leave it up to you.

Reply:
@ok2c  please do anther pass

Reply:
@arturobernalg I am not sure I see what has changed. `EnvironmentProxyConfigurer#apply` call is still there instead of “do nothing unless asked”.